apps/mobile/src/hooks/balances/use-stx-balances.ts

```diff
@@ -0,0 +1,41 @@
+import { MockedAccount } from '@/mocks/account.mocks';
+import { useStacksSigners } from '@/store/keychains/stacks/stacks-keychains.read';
+
+import { Money } from '@leather.io/models';
+import {
+  useCryptoCurrencyMarketDataMeanAverage,
+  useStxCryptoAssetBalance,
+} from '@leather.io/query';
+import { baseCurrencyAmountInQuote, createMoney, sumMoney } from '@leather.io/utils';
+
+interface StxBalance {
+  stxBalance: Money;
+  stxBalanceUsd: Money;
+}
+
+function useGetAccountStxBalance(account: MockedAccount): Money {
+  const signers = useStacksSigners().fromAccountIndex(account.fingerprint, account.accountIndex);
+  // TODO - ask Kyran if the magic number 0 is correct - signers always returns an array
+  const stxAddress = signers[0]?.address;
+  const {
+    filteredBalanceQuery: { data: balance },
+  } = useStxCryptoAssetBalance(stxAddress ?? '');
+
+  return balance ? balance.totalBalance : createMoney(0, 'STX');
+}
+
+function useGetAccountsStxBalance(accounts: MockedAccount[]): Money {
+  if (!accounts.length) return createMoney(0, 'STX');
+
+  return accounts.reduce(
+    (total, account) => sumMoney([total, useGetAccountStxBalance(account)]),
```

Comment:
I changed this to avoid the error: ` "useGetAccountStxBalance" cannot be called inside a callback. React Hooks must be called in a React function component or a custom React Hook function  react-hooks/rules-of-hooks` 

We have a lot of different hooks, calling each other and without refactoring a lot of code I can solve this issue by simply using a React function component - AKA calling the same function `GetAccountStxBalance` instead of `useGetAccountStxBalance`. 

I know we have rules about naming so I'm not sure if the team agrees on this but it seems like a reasonable way to avoid this error

Reply:
This is outdated and replaced by `useQueries` - https://github.com/leather-io/mono/pull/542#issuecomment-2431713706